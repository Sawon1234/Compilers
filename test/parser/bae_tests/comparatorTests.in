
fun testLesssThan(Boolean a, Boolean b) : Boolean
{
	return a<b;
}

fun testLessThanEq(Integer a, Integer b) : Boolean
{
	return a<=b;
}

fun testGreaterThanEq(Integer a, Integer b): Boolean
{
	return a>=b;
}

fun testGreaterThan(Integer a, Integer b) : Boolean
{
	return a>b;
}

fun testEq(Integer a, Integer b) : Boolean
{
	return a==b;
}

fun testNotEq(Integer a, Integer b) : Boolean
{
	return a!=b;
}