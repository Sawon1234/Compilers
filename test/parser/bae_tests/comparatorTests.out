fun testLesssThan ( a : Boolean < > , b : Boolean < > ) : Boolean {	return a . lessThan < >( b , true ) ; } fun testLessThanEq ( a : Integer < > ,  b : Integer < > ) : Boolean { return a . lessThan < > ( b , false ) ; } fun testGreaterThanEq ( a : Integer < > , b : Integer < > ) : Boolean { return b . lessThan < > ( a , true ) ; } fun testGreaterThan ( a : Integer < > , b : nteger < > ) : Boolean { return b . lessThan < > ( a ) ; } fun testEq ( a : Integer < > , b : Integer < > ) : Boolean { return a . equals ( b ) ; } fun testNotEq ( a : Integer < > , b : Integer < > ) : Boolean { return a . equals negate ( b ) ; }